base/intfuncs.jl

```diff
@@ -755,9 +755,15 @@ function digits!(a::AbstractVector{T}, n::Integer; base::Integer = 10) where T<:
     isempty(a) && return a
 
     if base > 0
-        for i in eachindex(a)
-            n, d = divrem(n, base)
-            a[i] = d
+        if base == 2
+            for i in eachindex(a)
+                a[i] = (n >> (i - 1)) & 1
```

Comment:
Unfortunately it doesn't work with offset arrays. I'm not certain we should bother with these, but for now we do, and this is breaking. You would need to replace the second `i` in this line with something like `i - firstindex(x) + 1`.

Reply:
You mean `i - firstdindex(a)`?  I don't get the `+ 1`.

Reply:
The `+1` I mentioned is canceled by the existing `-1` in this line.